### ISSUE TITLE
Implement custom HTML reporter

### ISSUE THREAD

**Issue by SaladFork** [switchfly/ember-cli-mocha#71]
Currently `ember-cli-mocha` just uses the HTML reporter that comes with mocha. While it works very well, it is difficult to extend and does not offer the same features as the QUnit output does. I came across @ef4's [better-mocha-html-reporter](https://github.com/ef4/better-mocha-html-reporter) which implemented an analogous reporter in ES6 and also cleaned up the styling.

I've since [extended it a bit further](https://gist.github.com/SaladFork/15683b00388bfe1d1458) to add features such as hiding passed tests, changing the favicon to indicate the final test result, and group together JSHint results. I've been sharing this through gists and had help from @elbeezy in [cleaning it up](https://gist.github.com/elbeezy/d7ab4dda1bea033d646a).

Would such a reporter have a place in the `ember-cli-mocha` addon? Hoping this issue can start a discussion about the potential inclusion of a custom reporter (and if so, what features to implement).

**Comment by ballPointPenguin** [comment]
I agree that a general pattern for extending ember-cli-mocha with custom reporters would be helpful, but I also think that your reporter (or some agreed-upon variation of the @ef4 "better" reporter) should be the new default.

**Comment by ballPointPenguin** [comment]
see https://github.com/ballPointPenguin/ember-cli-mocha/commit/9ebb73f2a8c874c2afaacf73c7b3c6f1a1b66ce8

**Comment by mmelvin0** [comment]
I combined the work of @ef4, @SaladFork, and @lolmaus (and perhaps others) into a proper Ember-CLI addon as I found the default Mocha test runner to be a bit lacking. Feedback is welcomed.

https://github.com/mmelvin0/ember-cli-mocha-reporter

**Comment by ballPointPenguin** [comment]
thanks @mmelvin0 this has been fermenting on the dark backburner of my mind
